core/domain/suggestion_services_test.py

```diff
@@ -2084,6 +2094,1155 @@ def test_create_translation_contribution_stats_from_model(self) -> None:
+        suggestion_services.update_question_review_stats(
+            suggestion_services.get_suggestion_by_id(
+                initial_suggestion.suggestion_id)
+        )
+        suggestion_services.update_question_review_stats(
+            suggestion_services.get_suggestion_by_id(
+                latest_suggestion.suggestion_id)
+        )
+        question_review_stats_model = (
+            suggestion_models.QuestionReviewStatsModel.get(
+                self.reviewer_id, topic_id
+            )
+        )
+        # Assert question review stats after the review.
+        # At this point we can confirm that there should be an associated
+        # question review stat object for the given IDs since we have
+        # called update_question_review_stats function to create/update question
+        # review stats.
+        assert question_review_stats_model is not None
+        self.assertEqual(
+            question_review_stats_model.accepted_questions_count,
+            2
+        )
+        self.assertEqual(
+            (
+                question_review_stats_model
+                .reviewed_questions_count
+            ),
+            2
+        )
+
+    def test_update_question_stats_when_suggestion_is_accepted(
+        self) -> None:
+        # This test case will check stats of the reviewer and the submitter
+        # when a question suggestion is accepted.
+        # Steps required in the setup phase before testing.
+        # 1. Save new skills.
+        # 2. Save a topic assigning skills for it.
+        # 4. Create a question suggestion.
+        skill_id_1 = self._create_skill()
+        skill_id_2 = self._create_skill()
+        topic_id = self._create_topic(skill_id_1, skill_id_2)
+        initial_suggestion = self._create_question_suggestion(skill_id_1)
+        latest_suggestion = self._create_question_suggestion(skill_id_2)
+        suggestion_services.update_question_contribution_stats_at_submission(
+            initial_suggestion
+        )
+        suggestion_services.update_question_contribution_stats_at_submission(
+            latest_suggestion
+        )
+
+        question_contribution_stats_model = (
+            suggestion_models.QuestionContributionStatsModel.get(
+                self.author_id, topic_id
+            )
+        )
+        # Assert question contribution stats before the review.
+        # At this point we can confirm that there should be an associated
+        # question contribution stat object for the given IDs since we have
+        # called update_question_contribution_stats_at_submission function to
+        # create/update question contribution stats.
+        assert question_contribution_stats_model is not None
+        self.assertEqual(
+            question_contribution_stats_model.submitted_questions_count,
+            2
+        )
+        self.assertEqual(
+            question_contribution_stats_model.accepted_questions_count,
+            0
+        )
```

Comment:
Seems a bit redundant since we've already tested this part in the test above. I would remove these lines here and in the tests below.

Reply:
Actually it is added as a base verification of contributor stats before it is reviewed.

Reply:
Right, but we've already tested the behavior when submitting in the test above, so it doesn't seem necessary to repeat it here, when the main thing we're testing in this test is the behavior after accepting a suggestion. Ditto below.

Reply:
Done